src/ttt_core/console.clj

```diff
@@ -59,50 +65,73 @@
 (defn draw-main [game message]
   (do
     (clear-terminal)
-    (draw-header)
+    (draw-header (:language game))
     (draw-player-info game)
     (draw-board (:board game))
     (println message)
     (draw-footer)))
 
-(defn build-choose-game-mode-string 
-  ([] (build-choose-game-mode-string ""))
-  ([message]
-    (str message "Choose a game mode:\n1. Human-vs-Human\n2. Human-vs-Computer")))
-
-(defn request-game-mode
-  ([game] (request-game-mode game ""))
-  ([game message]
-    (let [modes {:1 :human-vs-human :2 :human-vs-computer}]
+(defn build-selection-string 
+  ([language msg-keyword] (build-selection-string language msg-keyword ""))
+  ([language msg-keyword prepend-msg]
+    (str prepend-msg
+      (if (= "" prepend-msg) "" " ")
+      (translate [language] [msg-keyword]))))
+        
+(defn request-game-setup
+  ([game] (request-game-setup game ""))
+  ([game prepend-msg]
+    (let [opts {:1 :human-vs-human :2 :human-vs-computer :3 :language-setup}]
+      (do
+        (draw-main game (translate [(:language game)] [:game-setup-selection-header]))
+        (println (build-selection-string (:language game) :choose-game-setup prepend-msg))
+        (get opts (keyword (read-line)))))))
+
+(defn request-language-setup
+  ([game] (request-language-setup game ""))
+  ([game prepend-msg]
+    (let [opts {:1 :en :2 :pl}]
       (do
-        (draw-main game "--- Game Mode Selection ---")
-        (println (build-choose-game-mode-string message))
-        (get modes (keyword (read-line)))))))
+        (draw-main game (translate [(:language game)] [:game-language-selection-header]))
+        (println (build-selection-string (:language game) :choose-language-setup prepend-msg))
+        (get opts (keyword (read-line)))))))
 
-(defn get-game-mode-selection [game] 
-  (loop [selection (request-game-mode game)]
+(defn get-selection [game request-fn] 
```

Comment:
Similar to the `build-selection-string` function, `get-selection` is meant to DRY the code by taking a `request-fn` parameter which determines what kind of input (language, game-mode, etc) is being requested and then getting / validating it from the user.